python/ray/dataframe/dataframe.py

```diff
@@ -748,7 +750,9 @@ def last(self, offset):
         raise NotImplementedError("Not Yet implemented.")
 
     def last_valid_index(self):
-        raise NotImplementedError("Not Yet implemented.")
+        return max(
```

Comment:
Same here regarding the `ray.Index` object.

Reply:
resolved by getting self._index[-1]